Add Episode 5 Lesson Text (#49)
czi-catalystproject/hub-champion-training: episodes/transfer_data.md

(episode-data)=
# Navigating the filesystem and transferring data to and from the hub

> teaching: 30
> exercises: 20

Author: ```suggestion
<!-- teaching: 30
exercises: 20 -->
```

::::

## Shared directory

:::{figure} ../media/episodes/transfer_data/shared.png
:alt: The shared-readwrite folder appears for Hub Champions only.
:figclass: margin

The *shared-readwrite* folder appears for Hub Champions only.
:::

> All hub users have a access to a *shared* directory from their home directory. This is a read-only directory where users can access large files or datasets for their work and its contents cannot be accidentally modified or deleted.

Reviewer: ```suggestion
All hub users have access to a common *shared* directory from their home directory. This is a read-only directory where users can access large files or datasets for their work and its contents cannot be accidentally modified or deleted.
```

> Only Hub Champions can populate the *shared directory* by placing files in the *shared-readwrite* directory that appears in the home directory of Hub Champions only.

Reviewer: ```suggestion
Hub Champions can populate the *shared directory* by placing files in the *shared-readwrite* directory that appears only in their home directory.
```

:::

## Share and manage code with GitHub

> GitHub is a popular platform for creating, storing, managing and sharing code with collaborators. For general guides for using the Git extensions for JupyterLab and RStudio, please refer to [here](https://blog.reviewnb.com/jupyterlab-git-extension/#git-workflow) and [here](https://docs.posit.co/ide/user/ide/guide/tools/version-control.html#integrations), respectively.

Reviewer: I think this would benefit from another sentence, between the two already here, that explains what Git. (Can be brief, and limited to the context of GitHub and hub management.)

Author: ```suggestion
GitHub is a popular platform for creating, storing, managing and sharing code with collaborators. This is useful for synchronising local code development on a hub with a remote GitHub repository. Known as version control, this system allows users to track changes to code over time and can facilitate collaborative work across different machines. For general guides for using the Git extensions for JupyterLab and RStudio, please refer to [here](https://blog.reviewnb.com/jupyterlab-git-extension/#git-workflow) and [here](https://docs.posit.co/ide/user/ide/guide/tools/version-control.html#integrations), respectively.
```

Reviewer: Inspired by your start, I came up with this. What do you think?

```suggestion
GitHub is a popular platform for creating, storing, managing and sharing code with collaborators. Code developed on a hub can be synchronised with a remote repository on GitHub, and changes tracked in a process known as version control. This allows users to compare and even restore different versions of the code, and facilitates collaborative work across different machines. Version control and synchronisation of code with GitHub is enabled by a popular tool called Git. For general guides for using the Git extensions for JupyterLab and RStudio, please refer to [here](https://blog.reviewnb.com/jupyterlab-git-extension/#git-workflow) and [here](https://docs.posit.co/ide/user/ide/guide/tools/version-control.html#integrations), respectively.
```

Reviewer: Finally, the link text in this paragraph is not very accessible: [we should avoid links with undescriptive text like "here", "this page", etc](https://webaim.org/techniques/hypertext/link_text#text).

Perhaps you can replace with something like: 

> On a hub, version control of code is facilitated by Git extensions to the development environment. Learn more about how to use these:
> - [Git extension for JupyterLab](https://blog.reviewnb.com/jupyterlab-git-extension/#git-workflow)
> - [Git extension for RStudio](https://docs.posit.co/ide/user/ide/guide/tools/version-control.html#integrations)

Author: ```suggestion
GitHub is a popular *hosting platform* for creating, storing, managing and sharing code with collaborators. Code developed on a hub can be synchronised with a remote repository on GitHub, and changes are tracked in a process known as version control. This allows users to compare and even restore different versions of the code and facilitates collaborative work across different machines. Version control and synchronisation of code with GitHub is enabled by a popular tool called Git. On a hub, version control of code is facilitated by Git extensions to the development environment. Learn more about how to use these:

 - [Git extension for JupyterLab](https://blog.reviewnb.com/jupyterlab-git-extension/#git-workflow)
 - [Git extension for RStudio](https://docs.posit.co/ide/user/ide/guide/tools/version-control.html#integrations).
```

Reviewer: Excellent 👍

:::{note}
We recommend performing git operations using the HTTPS protocol and not SSH, since on the hub HTTPS will always work behind a firewall or proxy.
:::

> When a user pulls and pushes code to a remote repository on GitHub, they will be prompted to provide their GitHub credentials.

Reviewer: There is quite a leap implied here: if the reader understands what is meant by "push", "pull", and "remote repository" here -- and understands why we would want to do this with code on a hub -- we can assume that they do not need the "what is GitHub" sentence in the previous paragraph...
